mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -1148,11 +1153,10 @@ impl TendermintCoin {
             .map_to_mm(|e| TendermintCoinRpcError::InvalidResponse(format!("balance is not u64, err {}", e)))
     }
 
-    #[allow(clippy::result_large_err)]
-    pub(super) fn account_id_and_pk_for_withdraw(
+    pub(super) fn extract_account_id_and_private_key(
         &self,
         withdraw_from: Option<WithdrawFrom>,
-    ) -> Result<(AccountId, Option<H256>), WithdrawError> {
+    ) -> Result<(AccountId, Option<H256>), String> {
```

Comment:
Are we not generally striving to avoid String errors (in favour of always typed errors)?

Reply:
This is an unrecoverable internal error. TBH I don't think creating types for this kind of places helps us to maintain the project, in my opinion it even makes it worse. But I am fine with any alternative option. Would you be okay with using `io::Error`?

Reply:
btw we don't like to use things like anyhow::Error?

Reply:
> TBH I don't think creating types for this kind of places helps us to maintain the project, in my opinion it even makes it worse.

I agree, that's why I passed on this on the review.

> Are we not generally striving to avoid String errors (in favour of always typed errors)?

Yes, we were striving to avoid String Errors in the past but we will refactor to a different error handling approach anyways.